pyccel/parser/syntactic.py

```diff
@@ -729,13 +730,18 @@ def fill_types(ls):
 
                 txt  = '#$ header template ' + str(tp_name)
                 txt += '(' + '|'.join(types) + ')'
-                if tp_name in templates:
+                if tp_name in tmp_templates:
                     msg = 'The template "{}" is duplicated'.format(tp_name)
                     errors.report(msg,
                                 bounding_box = (stmt.lineno, stmt.col_offset),
                                 severity='warning')
+                tmp_templates[tp_name] = hdr_parse(stmts=txt)
+                template['templates_dict'] = tmp_templates
```

Comment:
Shouldn't this be in an `else`?

Reply:
Why ?

Reply:
So you don't parse the header multiple times? Or is the aim to keep the last version?

Reply:
i think we don't need an else statement because the if block is raising a fatal error.

Reply:
I don't see the fatal error can you link me to the line? I see a warning, but a warning doesn't stop the execution

Reply:
sorry i mentioned the wrong if error, So i think its better without else, since we warn the user that a template is duplicated, what do you think ?

Reply:
ok. Keeping the last template probably makes sense :+1: